### PR TITLE
Consent communication channel placeholders

### PR DESCRIPTION
**Pull request by walaura**
## What does this change?
Adds a list of communication channels (literally, a list) to the consent journey and the contact settings page. As well as the method to dynamically show and hide them based on the users contact details. This is all under a switch

## What is the value of this and can you measure success?
GDPR compliance

## Screenshots
screen shot 2017-12-19 at 12 26 47

### CODE REVIEW

File: identity/app/views/support/fragment/ConsentChannel.scala [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package views.support.fragment
```
Comment (mario-galic) [inline identity/app/views/support/fragment/ConsentChannel.scala]: We use 2 spaces indentation in our Scala code.

File: identity/app/views/support/fragment/ConsentChannel.scala [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    def checkboxName(behaviour: ConsentChannelBehaviour): String = {
+        behaviour match {
+            case TextConsentChannel => "Text"
+            case PhoneConsentChannel => "Phone"
+            case PostConsentChannel => "Post"
+        }
+    }
+
+    def communicationChannelsForUser(user: com.gu.identity.model.User): List[ConsentChannelBehaviour] = {
```
Comment (mario-galic) [inline identity/app/views/support/fragment/ConsentChannel.scala]: You could import `com.gu.identity.model.User` and then just write 

```def communicationChannelsForUser(user: User): List[ConsentChannelBehaviour]```

**Review by mariogalic — APPROVED** [review]

File: identity/app/views/support/fragment/ConsentChannel.scala (hunk 1 above)
Comment (walaura) [reply identity/app/views/support/fragment/ConsentChannel.scala]: huh not sure how that slipped by

File: identity/app/views/consentJourney.scala.html [hunk 3]
```diff
@@ -27,6 +30,31 @@
     }
 }
 
+@channelStep() = {
+    @if(IdentityShowCommunicationChannelConsents.isSwitchedOn && communicationChannelsForUser(user).nonEmpty) {
```
Comment (mario-galic) [inline identity/app/views/consentJourney.scala.html]: Should the switch be on its own line?

```
if(IdentityShowCommunicationChannelConsents.isSwitchedOn) {
  if(communicationChannelsForUser(user).nonEmpty) {
  }
}
...
```

File: identity/app/views/profile/privacyForm.scala.html [hunk 4]
```diff
@@ -64,6 +67,16 @@ <h2 class="manage-account-header">Would you like to receive information from the
     </form>
 }
 
+@channelConsentForm = {
+    <div class="manage-account__switches">
+        <ul>
+            @communicationChannelsForUser(user).map((channel) =>
+                <li>{s"${checkboxName(channel)}"}</li>
```
Comment (mario-galic) [inline identity/app/views/profile/privacyForm.scala.html]: Maybe this instead:

```
@communicationChannelsForUser(user).map((channel) => Html(s"<li>${checkboxName(channel)}</li>"))
```

**Pull request merged**